### PR TITLE
Crime Assist pda program

### PR DESCRIPTION
**Pull request by Timfa2112**
## About the PR

Add a Crime Assist program, which helps Security and Lawyer players determine what crime was committed without having to have a wiki open.

## Why / Balance

For many players, evidently, reading up on Space Law is an insurmountable obstacle, and as a result they have trouble identifying what is and isn't a crime. For this reason, a PDA program that gives a basic way to see what sort of crime was committed may help.

## Technical details

The PDA program is fairly simple. It has a ew buttons (Start, Home, Yes, No) which are visible/invisible depending on the program's State. This State also determines which Localization string is displayed.

## Media

image
image
image
image
image
image
image
image

Base flowchart (though I fixed the issue where you're always guilty of Hooliganism)
image
Credit goes to @IcedQuinn for the chart! It's been both very helpful as well as the inspiration for the entire program.

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- Add: Crime Assist program to Security and Lawyer PDA's, to help identify a crime with a simple yes/no question tree.

### CODE REVIEW

**Comment by Timfa2112** [comment]
Draft because I still need to figure out how to add icons

**Comment by IcedQuinn** [comment]
For the record, I made the chart. Though I bless it's usage 👍 

I didn't really think about innocent being in there because you wouldn't be sentenced by a warden if you were. But it's a pretty hilarious thought.

**Comment by IcedQuinn** [comment]
After talking with Leo in Discord, I've drawn some (more acceptable) candidate sprites also:

space law icon 2 scales
space law icon 2 cuff
space law icon 2

**Comment by leonardo-dabepis** [comment]
> After talking with Leo in Discord, I've drawn some (more acceptable) candidate sprites also:
> 
> space law icon 2 scales space law icon 2 cuff space law icon 2

I said I would rather have the sprite just be the base space law book, just to keep it consistent.

EDIT: this isn't too big of a deal though. The first sprite on your list is preferable if you really insist on not using the base space law book.

**Comment by Timfa2112** [comment]
> > After talking with Leo in Discord, I've drawn some (more acceptable) candidate sprites also:
> > space law icon 2 scales space law icon 2 cuff space law icon 2
> 
> I said I would rather have the sprite just be the base space law book, just to keep it consistent.
> 
> EDIT: this isn't too big of a deal though. The first sprite on your list is preferable if you really insist on not using the base space law book.

I happen to have picked the first sprite in the list. I like this one much more than the simplistic book in the game, and I can see an application having a more elaborate book icon than you'd generally see in real life. 

That, and I like the way it looks :P

**Comment by Timfa2112** [comment]
Seeing a strange error in the 'Build & Test Debug' step; `Caught exception while generating mail: Entity HandcuffsDummy (21731, HandcuffsDummy) is/has been terminated`
Having a hard time seeing how any of my changes can cause such a thing, and it didn't happen in the previous run (while I only swapped a .png compared to then). Is it possible the test fails on this randomly due to a bug unrelated to this PR?

**Comment by Colin-Tel** [comment]
> Seeing a strange error in the 'Build & Test Debug' step; `Caught exception while generating mail: Entity HandcuffsDummy (21731, HandcuffsDummy) is/has been terminated` 

it’s unrelated. It has to do with mail tests that are a little finicky. Test packaging has also been failing on all PRs so I wouldn’t worry about it either

**Comment by Timfa2112** [comment]
I've fiddled with the formatting a bit more, and I think this is about as good as I'll make it. I'll update the images in the description, and then I think it's ready for review :)

File: Resources/Locale/en-US/deltav/cartridge-loader/cartridges.ftl [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+crime-assist-question-wassomeoneattacked = Was an entity attacked?
+crime-assist-question-wasitsophont = Was the victim in question a sophont?
+crime-assist-question-didvictimdie = Did the victim die as a result to the attack?
+crime-assist-question-isvictimremovedfrombody = Is the victim alive, but intentionally and permanently removed from the body?
+crime-assist-question-wasdeathintentional = Did the suspect attack the victim with intent to kill?
+crime-assist-question-forcedmindbreakertoxin = Was the victim forced to take Mindbreaker Toxin?
+crime-assist-question-hadillegitimateitem = Did the suspect have an illegitimate or controlled item, substance or entity?
+crime-assist-question-wasitaperson = Did the suspect hold another sophont against its will?
+crime-assist-question-wassuspectselling = Was the suspect selling or distributing the illegitimate items?
+crime-assist-question-wassuspectseentaking = Can it be proved that the suspect took the item knowing it was illegitimate for the suspect to possess?
```
Comment (Colin-Tel) [inline Resources/Locale/en-US/deltav/cartridge-loader/cartridges.ftl]:
```suggestion
crime-assist-question-hadillegitimateitem = Did the suspect have an illicit or controlled item, substance or entity?
crime-assist-question-wasitaperson = Did the suspect hold another sophont against its will?
crime-assist-question-wassuspectselling = Was the suspect selling or distributing the illicit items?
crime-assist-question-wassuspectseentaking = Can it be proved that the suspect took the item knowing it was illegal for the suspect to possess?
```
I may be tripping, but shouldn't this say "illicit" or "illegal" in these cases? "Illegitimate" sounds like a case of counterfeiting

**Review by Colin-Tel — CHANGES_REQUESTED** [review]
Please add this program to the following PDAs:
```
CorpsmanPDA
PrisonGuardPDA
SeniorOfficerPDA
```

File: Resources/Locale/en-US/deltav/cartridge-loader/cartridges.ftl (hunk 1 above)
Comment (Timfa2112) [reply Resources/Locale/en-US/deltav/cartridge-loader/cartridges.ftl]: You're right, that does sound better. Thanks for the suggestion!

**Comment by Colin-Tel** [comment]
let me know when you add it to the other PDAs, ~~I'll merge it afterwards.~~
My mistake, Debug should take a look at it since it touches code.

**Comment by Timfa2112** [comment]
> let me know when you add it to the other PDAs, ~I'll merge it afterwards.~ My mistake, Debug should take a look at it since it touches code.

It's been added to the PDA's! I'll await Debug's judgement now :)

**Comment by Timfa2112** [comment]
> Please add this program to the following PDAs:
> 
> ```
> CorpsmanPDA
> PrisonGuardPDA
> SeniorOfficerPDA
> ```

Added!

**Comment by IcedQuinn** [comment]
> > I said I would rather have the sprite just be the base space law book, just to keep it consistent.
>
> I happen to have picked the first sprite in the list. I like this one much more than the simplistic book in the game, and I can see an application having a more elaborate book icon than you'd generally see in real life.

I do think it looks better than the base sprite, and maybe the base sprite should be changed to remain consistent.

I won't be too offended if someone else wants to splice the original book sprite though.

**Review by Colin-Tel — DISMISSED** [review]
Cool addition! I'm sure newer Sec players will love to use this. Maybe consider adding an entry into the tips or something.

**Comment by Timfa2112** [comment]
> Cool addition! I'm sure newer Sec players will love to use this. Maybe consider adding an entry into the tips or something.

If I can, I'd love to! Though I don't know how to do that yet, so if you can point me in the right direction I'd appreciate it :)

**Comment by Colin-Tel** [comment]
> If I can, I'd love to! Though I don't know how to do that yet, so if you can point me in the right direction I'd appreciate it :)

should be here
https://github.com/DeltaV-Station/Delta-v/blob/master/Resources/Prototypes/Datasets/tips.yml

**Comment by Timfa2112** [comment]
> > If I can, I'd love to! Though I don't know how to do that yet, so if you can point me in the right direction I'd appreciate it :)
> 
> should be here https://github.com/DeltaV-Station/Delta-v/blob/master/Resources/Prototypes/Datasets/tips.yml

I've added one at the bottom of the file :)

**Comment by DebugOk** [comment]
I'll do a proper review of this sometime this week probably

**Comment by Timfa2112** [comment]
> I'll do a proper review of this sometime this week probably

Much appreciated, I'll await it patiently :)

File: Content.Client/DeltaV/CartridgeLoader/Cartridges/CrimeAssistUiFragment.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            CrimeAssistUiState.UiStates.Result_ObstructionOfJustice => "[color=#7b5430]" + Loc.GetString("crime-assist-crimetype-felony") + "[/color]",
+            CrimeAssistUiState.UiStates.Result_Murder => "[color=#7b2e30]" + Loc.GetString("crime-assist-crimetype-capital") + "[/color]",
+            CrimeAssistUiState.UiStates.Result_Terrorism => "[color=#7b2e30]" + Loc.GetString("crime-assist-crimetype-capital") + "[/color]",
+            CrimeAssistUiState.UiStates.Result_GrandSabotage => "[color=#7b2e30]" + Loc.GetString("crime-assist-crimetype-capital") + "[/color]",
+            CrimeAssistUiState.UiStates.Result_Decorporealisation => "[color=#7b2e30]" + Loc.GetString("crime-assist-crimetype-capital") + "[/color]",
+            CrimeAssistUiState.UiStates.Result_Kidnapping => "[color=#7b2e30]" + Loc.GetString("crime-assist-crimetype-capital") + "[/color]",
+            CrimeAssistUiState.UiStates.Result_Sedition => "[color=#7b2e30]" + Loc.GetString("crime-assist-crimetype-capital") + "[/color]",
+            CrimeAssistUiState.UiStates.Result_SexualHarrassment => "[color=#7b2e30]" + Loc.GetString("crime-assist-crimetype-capital") + "[/color]",
+            _ => ""
```
Comment (DebugOk) [inline Content.Client/DeltaV/CartridgeLoader/Cartridges/CrimeAssistUiFragment.xaml.cs]: It'd be nice if this functioned via yaml or something, as if we ever change splaw the only way to update this is by touching code

Comment (Timfa2112) [reply Content.Client/DeltaV/CartridgeLoader/Cartridges/CrimeAssistUiFragment.xaml.cs]: I would need to learn a bit more about yml for that, but it does sound like a big improvement. I should have time to look into this saturday :)

Comment (Timfa2112) [reply Content.Client/DeltaV/CartridgeLoader/Cartridges/CrimeAssistUiFragment.xaml.cs]: It is done! Learnt a lot about prototypes while doing it, too

**Comment by dvir001** [comment]
Cant wait to have this added.

**Review by DebugOk — APPROVED** [review]
I only glanced over it

**Pull request merged**